dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/v2/DataUploadRunnable.kt

```diff
@@ -125,12 +153,11 @@ internal class DataUploadRunnable(
         storage.confirmBatchRead(batchId, removalReason) {
             if (status.shouldRetry) {
```

Comment:
minor: then the whole lambda can be simplified as `it.markAsRead(deleteBatch = !status.shouldRetry)`